### PR TITLE
Prepare Uploader Vueification by removing jQuery

### PR DESCRIPTION
**Pull request by guerler**
Removal of jQuery and ajax calls from uploader utilities, use OpenAPI-fetcher instead. Consolidation of utilities.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/components/Upload/utils.js [hunk 1]
```diff
@@ -1,7 +1,149 @@
...
+        }
+        return a.text > b.text ? 1 : a.text < b.text ? -1 : 0;
+    };
+}
+
+async function loadDbKeys() {
+    if (_cachedDbKeys) {
+        return Promise.resolve(_cachedDbKeys);
+    }
+    const url = `${getAppRoot()}api/genomes`;
```
Comment (mvdbeek) [inline client/src/components/Upload/utils.js]: Can you please use the typescript fetcher for the endpoints that are in FastAPI, so everything moved to here.

File: client/src/components/Upload/utils.js [hunk 2]
```diff
@@ -1,7 +1,128 @@
...
+        "This system will try to detect the file type automatically. If your file is not detected properly as one of the known formats, it most likely means that it has some format problems (e.g., different number of columns on different rows). You can still coerce the system to set your data to the format you think it should be.  You can also upload compressed files, which will automatically be decompressed.",
+};
+export const DEFAULT_DBKEY = "?";
+export const DEFAULT_EXTENSION = "auto";
+
+/**
+ * Local cache.
+ */
+let _cachedDatatypes;
+let _cachedDbKeys;
```
Comment (mvdbeek) [inline client/src/components/Upload/utils.js:21]: Instead of adding our own cache system it may be a better option to use the store right away ?

**Review by mvdbeek — APPROVED** [review]
Still draft ?

File: client/src/components/Upload/utils.js (hunk 2 above)
Comment (guerler) [reply client/src/components/Upload/utils.js:21]: Yes, I agree, I made a note on the corresponding issue to move this to a store in a follow-ups.

**Pull request merged**